packages/server/src/api/routes/public/applications.ts

```diff
@@ -92,7 +93,19 @@ write.push(
  *               application:
  *                 $ref: '#/components/examples/application'
  */
-write.push(new Endpoint("delete", "/applications/:appId", controller.destroy))
+write.push(
+  new Endpoint(
+    "delete",
+    "/applications/:appId",
+    controller.destroy
+  ).addMiddleware((ctx, next) => {
+    // set unpublish if the appId is production, so we don't delete the object store bucket
+    if (isProdAppID(ctx.params.appId)) {
```

Comment:
Nice one!

The only thing about this may be confusion on the users side, where they delete with a prod app id and still see the dev app. Our public api docs don't document anything around unpublishing.

To make the api as easy to use as possible, should we just do the unpublish and the delete in one step if the prod app id is provided?

i.e. we could have a wrapping public api controller that calls `destroy` two times
or we could just update the base functionality in the existing controller to do this